app/src/main/java/com/CameraActivity.java

```diff
@@ -32,6 +34,11 @@
         System.loadLibrary("opencv_java3");
     }
 
+    Executor singleThreadExecutor = Executors.newSingleThreadExecutor();
```

Comment:
It all begs for separate class&object for thread execution, instead of throwing this variables and functions to main camera activity. Honestly I think that Activities should be only routers to functional blocks of code, instead of delivering whole functionality inside.

Reply:
+1